src/main/java/com/google/research/bleth/simulator/Beacon.java

```diff
@@ -0,0 +1,59 @@
+package com.google.research.bleth.simulator;
+
+/** Tracing Simulation's Beacon, which moves on the board and transmits its unique static ID each round. */
+public class Beacon implements IBeacon {
+    static int beaconId = 0; // used for generating unique id for each beacon.
+
+    private final int id;
+    private final MovementStrategy movementStrategy;
+    private final Simulation simulation;
+
+    private Location realLocation; // the beacon's location on the board, changed each time the beacon moves.
+
+    /**
+     * Create new Beacon with consecutive serial number.
+     * @param id is a unique ID.
+     * @param initialLocation is the location on board where the agent is placed.
+     * @param movementStrategy determines how the agent moves.
+     * @param simulation is the world the agent lives in.
+     */
+    Beacon(int id, Location initialLocation, MovementStrategy movementStrategy, Simulation simulation) {
+        this.id = id;
+        realLocation = initialLocation;
+        simulation.getBoard().placeAgent(realLocation,this);
```

Comment:
Consider moving this to the factory: as I mentioned in an earlier comment - any logic called from a constructor might be dangerous.
(E.g., what will happen if Board.placeAgent() wants to print out the agent's details in a debug message? The agent is still half-constructed at this point, so anything referring to e.g. movementStrategy may produce an NPE...)